packages/core/src/providers/laravel-passport.ts

```diff
@@ -0,0 +1,144 @@
+ *     LaravelPassport({ baseUrl: LARAVEL_PASSPORT_BASE_URL, clientId: LARAVEL_PASSPORT_CLIENT_KEY, clientSecret: LARAVEL_PASSPORT_CLIENT_SECRET }),
+ *   ],
+ * })
+ * ```
+ *
+ * ### Resources
+ *  - [Laravel Passport Documentation](https://laravel.com/docs/11.x/passport)
+ *
+ *
+ * ### Notes
+ *
+ * By default, Auth.js assumes that the Laravel Passport provider is
+ * based on the [OAuth 2](https://www.rfc-editor.org/rfc/rfc6749.html) specification.
+ *
+ * :::tip
+ *
+ * The Laravel Passport provider comes with a [default configuration](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/laravel-passport.ts).
+ * To override the defaults for your use case, check out [customizing a built-in OAuth provider](https://authjs.dev/guides/configuring-oauth-providers).
+ *
+ * :::
+ *
+ * :::info **Disclaimer**
+ *
+ * If you think you found a bug in the default configuration, you can [open an issue](https://authjs.dev/new/provider-issue).
+ *
+ * Auth.js strictly adheres to the specification and it cannot take responsibility for any deviation from
+ * the spec by the provider. You can open an issue, but if the problem is non-compliance with the spec,
+ * we might not pursue a resolution. You can ask for more help in [Discussions](https://authjs.dev/new/github-discussions).
+ *
+ * :::
+ */
+export default function LaravelPassport<P extends LaravelPassportProfile>(
+    options: OAuthUserConfig<P> & {
+        baseUrl: string
+    },
+  ): OAuthConfig<P> {
+    return {
+        id: "laravel-passport",
+        name: "Laravel Passport",
+        type: "oauth",
+
+        authorization: {
+            url: `${options.baseUrl}/oauth/authorize`,
+            params: {
+                scope: "",
+            },
+        },
+        token: {
+            url: `${options.baseUrl}/oauth/token`,
+            async request({ params, provider }) {
+                const res = await fetch(provider.token?.url!, {
+                    method: "POST",
+                    headers: {
+                        "Cache-Control": "no-cache",
+                        "Content-Type": "application/x-www-form-urlencoded",
+                    },
+                    body: new URLSearchParams({
+                        grant_type: "authorization_code",
+                        client_id: provider.clientId!,
+                        client_secret: provider.clientSecret!,
+                        redirect_uri: provider.callbackUrl!,
+                        code: params.code!,
+                    }),
+                }).then((res) => res.json());
+
+                const tokens: TokenSet = {
+                    access_token: res.access_token,
+                    expires_at: res.expires_in,
+                    refresh_token: res.refresh_token,
+                    scope: res.scope,
+                    token_type: res.token_type,
+                }
+
+                return {
+                    tokens,
+                }
+            }
```

Comment:
The token request function https://github.com/nextauthjs/next-auth/pull/11057#pullrequestreview-2092593912, also i see that there's really no need to take control of the token request since the provider is pretty much following the OAUTH standards.
```suggestion
```